mevislab.github.io/content/tutorials/visualization/visualizationexample7.md

```diff
@@ -0,0 +1,61 @@
+---
+title: "Example 7: Add 3D viewer to OrthoView2D"
+date: 2023-11-21
+status: "OK"
+draft: false
+weight: 590
+tags: ["Beginner", "Tutorial", "Visualization", "3D", "OrthoView2D"]
+menu: 
+  main:
+    identifier: "visualization_example7"
+    title: "Add 3D viewer to OrthoView2D."
+    weight: 590
+    parent: "visualization"
+---
+# Example 7: Add 3D viewer to OrthoView2D {#TutorialVisualizationExample7}
+
+## Introduction
+In this example we like to use the `OrthoView2D` module and add a 3D viewer to the layout *Cube*.
+
+## Steps to do
+### Develop your network
+Add the modules `LocalImage` and `OrthoView2D` to your workspace and connect them.
+
+![Network](/images/tutorials/image_processing/network_example7.png "Network")
+
+The `OrthoView2D` module allows you to select multiple layouts. Select layout *Cube Equal*. The layout shows your image in three orthogonal viewing directions. The top left segment remains empty.
+
+![OrthoView2D Layouts](/images/tutorials/image_processing/network_example7_2.png "OrthoView2D Layouts")
+
+We now want to use a 3D rendering in the top left segment, whenever the layout *Cube Equal* is chosen. Add a `View3D` and a `SoViewportRegion` module to your workspace. Connect the `LocalImage` with your `View3D`. The image is rendered in 3D. Hit {{< keyboard "SPACE" >}} on your keyboard to make the hidden output of the `View3D` module visible. Connect it with your `SoViewportRegion` and connect the `SoViewportRegion` with the *inInvPreLUT* input of the `OrthoView2D`.
+
+![Network with SoViewportRegion](/images/tutorials/image_processing/network_example7_3.png "Network with SoViewportRegion")
+
+Open the `OrthoView2D` and inspect your layout.
+
+![OrthoView2D with 3D](/images/tutorials/image_processing/network_example7_4.png "OrthoView2D with 3D")
+
+You can see your `View3D` being visible in the bottom right segment of the layout behind the coronal view of the image. Open the panel of the `SoViewportRegion` module. In section *X-Position and Width*, set *Left Border* to *0* and *Right Border* to *0.5*. In section *Y-Position and Height*, set *Lower Border* to *0* and *Upper Border* to *0.5*. Also check *Render delayed paths*.
+
+![Define viewport region](/images/tutorials/image_processing/network_example7_5.png "Define viewport region")
+
+The `View3D` image is now rendered to the top left segment of the `OrthoView2D` because the module `SoViewportRegion` renders a sub graph into a specified viewport region (VPR). The problem is: We cannot rotate and pan the 3D object, because there is no camera interaction anymore after adding the `SoViewportRegion`. The camera interaction is taken by the `View3D` module before the viewport is taken.
```

Comment:
because -> , because
anymore -> available
The camera interaction is consumed by the `View3D` module before it can be used by the viewport.